sidecar_acceptor.go

```diff
@@ -525,41 +525,23 @@ func (a *SidecarAcceptor) handleServerMessage(
 	switch msg := serverMsg.Msg.(type) {
 
 	case *auctioneerrpc.ServerAuctionMessage_Prepare:
-		batchID := msg.Prepare.BatchId
 
 		sdcrLog.Tracef("Received prepare msg from server, "+
-			"batch_id=%x: %v", batchID, spew.Sdump(msg))
+			"batch_id=%x: %v", msg.Prepare.BatchId, spew.Sdump(msg))
 
-		newBatch, err := a.matchPrepare(a.pendingBatch, msg.Prepare)
-		if err != nil {
-			sdcrLog.Errorf("Error handling prepare message: %v",
-				err)
-			return a.sendRejectBatch(batchID, err)
+		if err := a.matchPrepare(msg.Prepare); err != nil {
+			sdcrLog.Errorf("unable to handle prepare message: %v", err)
+			return a.sendRejectBatch(msg.Prepare.BatchId, nil, err)
 		}
 
-		// We know we're involved in a batch, so let's store it for the
-		// next step.
-		a.pendingBatch = newBatch
```

Comment:
Why do we no longer store the batch here? We'll need to remove it if the execution fails for w/e reason.

Reply:
Yes, we do, but all the logic is inside the `a.matchPrepare`, `a.matchSign` and `a.matchFinalize`